src/api/altorenderer.cpp

```diff
@@ -0,0 +1,255 @@
+        snprintf(id_buffer, BUFSIZE - 1, "%s_%d", base.c_str(), num1);
+        id_buffer[BUFSIZE - 1] = '\0';
+        *alto_str += " ID=\"";
+        *alto_str += id_buffer;
+        *alto_str += "\"";
+    }
+
+    /**
+    * Add coordinates to specified TextBlock, TextLine, or String bounding box
+    * Add word confidence if adding to a String bounding box
+    */
+    static void AddBoxToAlto(const ResultIterator *it, PageIteratorLevel level,
+                             STRING *alto_str) {
+        int left, top, right, bottom;
+        it->BoundingBox(level, &left, &top, &right, &bottom);
+
+        int hpos = left;
+        int vpos = top;
+        int height = bottom - top;
+        int width = right - left;
+
+        *alto_str += " HPOS=\"";
+        alto_str->add_str_int("", hpos);
+        *alto_str += "\"";
+        *alto_str += " VPOS=\"";
+        alto_str->add_str_int("", vpos);
+        *alto_str += "\"";
+        *alto_str += " WIDTH=\"";
+        alto_str->add_str_int("", width);
+        *alto_str += "\"";
+        *alto_str += " HEIGHT=\"";
+        alto_str->add_str_int("", height);
+        *alto_str += "\"";
+
+        if (level == RIL_WORD) {
+            int wc = it->Confidence(RIL_WORD);
+            *alto_str += " WC=\"0.";
+            alto_str->add_str_int("", wc);
+            *alto_str += "\"";
+        }
+        if (level != RIL_WORD) {
+
+            *alto_str += ">";
+        }
+    }
+
+    /**
+     * Make an XML-formatted string with ALTO markup from the internal
+     * data structures.
+     */
+        char *TessBaseAPI::GetAltoText(int page_number) {
+            return GetAltoText(nullptr, page_number);
+        }
+
+    /**
+     * Make an XML-formatted string with ALTO markup from the internal
+     * data structures.
+     */
+        char *TessBaseAPI::GetAltoText(ETEXT_DESC *monitor, int page_number) {
+            if (tesseract_ == nullptr || (page_res_ == nullptr && Recognize(monitor) < 0))
+                return nullptr;
+
+            int lcnt = 0, bcnt = 0, wcnt = 0;
+            int page_id = page_number;
+
+            STRING alto_str("");
+
+            if (input_file_ == nullptr)
+                SetInputName(nullptr);
+
+    #ifdef _WIN32
+            // convert input name from ANSI encoding to utf-8
+          int str16_len =
+              MultiByteToWideChar(CP_ACP, 0, input_file_->string(), -1, nullptr, 0);
+          wchar_t *uni16_str = new WCHAR[str16_len];
+          str16_len = MultiByteToWideChar(CP_ACP, 0, input_file_->string(), -1,
+                                          uni16_str, str16_len);
+          int utf8_len = WideCharToMultiByte(CP_UTF8, 0, uni16_str, str16_len, nullptr, 0,
+                                             nullptr, nullptr);
+          char *utf8_str = new char[utf8_len];
+          WideCharToMultiByte(CP_UTF8, 0, uni16_str, str16_len, utf8_str,
+                              utf8_len, nullptr, nullptr);
+          *input_file_ = utf8_str;
```

Comment:
Could `*input_file` be used directly instead of `utf8_str` thus avoiding the copying and `delete[]`?

Reply:
I need help making this change. I did not write this section of code (It came from the hOCR implementation.), so I am not 100% sure of what it is doing.